scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,82 @@
+# 0010 - Separate Versioning Stream for Vulnerability Bundles
+
+- **Author(s):** J. Victor Martins <jvdm@sdf.org>
+- **Created:** [2024-07-22 Mon]
+
+## Status
+
+Status: Accepted
+
+Updates: [#0009](0009-scannerv4-read-manual-vulnerabilities.md)
+
+## Context
+
+A Vulnerability Bundle is an archive containing updates to vulnerability data in Scanner.  They are constantly updated by the Vulnerability Exporter (aka. Exporter), which runs periodically in GitHub's Workflows and uses ClairCore to fetch the security data to convert it to a JSON-based schema.  Then, the workflow bundles the data into a ZIP archive.  The schema is owned by ClairCore.
+
+A Scanner instance runs the Vulnerability Updater (aka. Updater) to fetch the bundle and use ClairCore to parse the schema and update the vulnerability database.  Suppose ClairCore changed the schema in a backward-incompatible manner, and that change is incorporate in the Scanner repository (a ClairCore version bump).  In that scenario, the next Exporter execution will update the bundle with the new schema.  Then, existing Scanner instances won't be able to consume the data.  A Scanner version bump to a version of ClairCore supporting the new schema would be required.
+
+To address that scenario, Scanner created one bundle per Stackrox release.  Each bundle is created with an Exporter running the same ClairCore version used to build the release, ensuring compatibility.  Bundles are served at `https://definitions.stackrox.io/v4/V/vulnerabilities.zip` where `V` will be a Stackrox release version such as "4.5.0".  Scanner uses its release version tagged during build time to determine which version to fetch from.  The versions to build bundles for are taken from a file named `scanner/updater/version/RELEASE_VERSION`.
+
+For its purpose, this approach needs to be revised.
```

Comment:
nit:
```suggestion
For the above reason(s), this approach needs to be revised.
```

Reply:
By "for its purpose" I meant "so it can fulfill its purpose and original goal". Is "the above reasons(s)" the same meaning as that?

Reply:
Similar meaning yes - I was confused when reading what does 'its' refer to from the above paragraphs.  Is 'its' referring to scanner? scanner definitions? vulnerabilities? the issues identified? - and what is 'purpose' referring to - to scan images? to create bundles? to load bundles? etc.

Up to you, to me the change reduces the ambiguity / is more clear.